### PR TITLE
libopenmpt 0.4.3 (new formula)

### PR DESCRIPTION
**Pull request by kavu**
Add new formula with libopenmpt, along with small utility openmpt123. 

libopenmpt is a library to play to play tracker music files (XM, MOD, etc.).

openmpt123 is a small player based on libopenmpt to demonstrate libopenmpt.

Based on #33005

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Only the last requirement (`audit`) left unchecked because `audit` asks to remove `:optional` and `:recommended` (`Formulae should not have optional or recommended dependencies`) but I am not quite sure how to allow build with SDL or SDL2 without specifying dependency scope.

### CODE REVIEW

**Comment by kavu** [comment]
Yeah, CI failed with exactly the same problem: 

```
17:57:44 Error: 8 problems in 1 formula detected
17:57:44 libopenmpt:
17:57:44   * Formulae should not have optional or recommended dependencies
17:57:44   * fully scope test system calls e.g. system "#{bin}/openmpt123"
```

Also, should I fix the second one with replacing `system "openmpt123", "--probe", "mystique.s3m"` with `system "#{bin}/openmpt123", "--probe", "mystique.s3m"`?

File: Formula/libopenmpt.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  desc "Software library to decode tracked music files"
+  homepage "https://lib.openmpt.org/libopenmpt/"
+  url "https://lib.openmpt.org/files/libopenmpt/src/libopenmpt-0.4.3+release.autotools.tar.gz"
+  version "0.4.3"
+  sha256 "d77443a279003921d6f0c4edb30d1e9dda387983f44113a6d58f623c1e6942ae"
+
+  depends_on "pkg-config" => :build
+
+  depends_on "portaudio"
+  depends_on "flac" => :recommended
```
Comment (fxcoudert) [inline Formula/libopenmpt.rb]: No options for formulas in Homebrew core. Either depend on it, or don't.

File: Formula/libopenmpt.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                          sndfile,
+                          sdl,
+                          sdl2
+
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system "curl", "-o", "mystique.s3m", "https://api.modarchive.org/downloads.php?moduleid=54144#mystique.s3m"
```
Comment (fxcoudert) [inline Formula/libopenmpt.rb]: Use a `resource` block for that file.

File: Formula/libopenmpt.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    system "#{bin}/openmpt123", "--probe", "mystique.s3m"
+    raise "Probing a known-good file failed" unless $CHILD_STATUS.success?
```
Comment (fxcoudert) [inline Formula/libopenmpt.rb]: This is unnecessary, `system` will fail if the exit code is nonzero.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/libopenmpt.rb (hunk 1 above)
Comment (kavu) [reply Formula/libopenmpt.rb]: Got it, thank you!

**Comment by kavu** [comment]
@fxcoudert, fixed everything and CI is now green.

File: Formula/libopenmpt.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                          "--prefix=#{prefix}",
+                          "--without-vorbisfile"
+
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    resource("mystique.s3m").stage do
+      system "#{bin}/openmpt123", "--probe", "mystique.s3m"
```
Comment (fxcoudert) [inline Formula/libopenmpt.rb]: Is there anything we can check in the output of this command?

Comment (kavu) [reply Formula/libopenmpt.rb]: Yeah, at least there is a "Success" substring in output. Can you advise how I can capture it?

Comment (fxcoudert) [reply Formula/libopenmpt.rb]:
```suggestion
      output = shell_output("#{bin}/openmpt123 --probe mystique.s3m")
      assert_match "Success", output
```

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @kavu for your contribution to Homebrew!

**Pull request closed without merging**